corehq/apps/tour/models.py

```diff
@@ -0,0 +1,22 @@
+from django.contrib.auth.models import User
+from django.db import models, IntegrityError
+
+
+class GuidedTour(models.Model):
+    user = models.ForeignKey(
+        User,
+        db_index=True,
+    )
+    tour_slug = models.CharField(
+        max_length=255,
+        db_index=True,
+    )
+    date_completed = models.DateTimeField(auto_now=True)
+
+    @classmethod
+    def has_seen_tour(cls, user, tour_slug):
+        return cls.objects.filter(user=user, tour_slug=tour_slug).count() > 0
+
+    @classmethod
+    def mark_as_seen(cls, user, tour_slug):
+        cls.objects.get_or_create(user=user, tour_slug=tour_slug)
```

Comment:
I think we should add a `unique_together` constraint for `user` + `tour_slug`, right? https://docs.djangoproject.com/en/1.9/ref/models/options/#unique-together